go/integration_test/processor/calls_test.go

```diff
@@ -193,20 +186,266 @@ func TestProcessor_DifferentCallTypesAccessStorage(t *testing.T) {
+						t.Errorf("%s did not set the correct sender, got unexpected output: %v", callName, result.Output)
+					}
+				}
+			})
+		}
+	}
+}
+
+func TestProcessor_RecursiveCallsAfterAStaticCallAreStatic(t *testing.T) {
+	gasLimit := tosca.Gas(100000000)
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue // todo implement different call types
+		}
+		calls := callTypesAndProperties()
+		for callName, call := range calls {
+			t.Run(fmt.Sprintf("%s-%s", processorName, callName), func(t *testing.T) {
+
+				// Test structure:
+				// sender0 --CALL--> receiver0 --STATICCALL--> receiver1
+				// --test.CALL--> receiver2 -> SSTORE
+				// after the first static calls all following calls have to be static,
+				// therefore the SSTORE should revert. The success boolean of the test.CALL
+				// is returned as output.
+
+				sender0 := tosca.Address{1}
+				receiver0 := tosca.Address{2}
+				receiver1 := tosca.Address{3}
+				receiver2 := tosca.Address{4}
+
+				code0 := pushCallArguments(calls["staticCall"], gasLimit, tosca.Value{}, receiver1)
+				// perform static call and forward the result
+				code0 = append(code0, []byte{
+					byte(vm.STATICCALL),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				code1 := pushCallArguments(call, gasLimit, tosca.Value{}, receiver2)
+				// perform call and return whether it was successful
+				code1 = append(code1, []byte{
+					byte(call.callType),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.MSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				// perform sstore (should revert in static mode)
+				code2 := []byte{
+					byte(vm.PUSH1), byte(0),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.SSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}
+
+				state := WorldState{
+					sender0:   Account{},
+					receiver0: Account{Code: code0},
+					receiver1: Account{Code: code1},
+					receiver2: Account{Code: code2},
+				}
+				scenario := Scenario{
+					Transaction: tosca.Transaction{
+						Sender:    sender0,
+						Recipient: &receiver0,
+						GasLimit:  gasLimit,
+					},
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, scenario.Transaction, transactionContext)
+				if err != nil || !result.Success {
+					t.Errorf("execution was not successful or failed with error %v", err)
+				}
+				if !slices.Equal(result.Output, bytes.Repeat([]byte{0}, 32)) {
+					t.Errorf("%s did not enforce static mode and returned %v", callName, result.Output)
+				}
+			})
+		}
+	}
+}
+
+func pushCallArguments(
+	call callProperties,
+	gasLimit tosca.Gas,
+	transferValue tosca.Value,
+	receiver tosca.Address,
+) []byte {
+	var code []byte
+	if call.hasValue {
+		code = pushToStack([]*big.Int{
```

Comment:
would it make sense to use `tosca.Word` instead of `*big.Int` ? there shouldn't be any impact, I just thought "why is big int needed here?"

Reply:
We had a discussion about this when I implemented the `pushToStack` function, due to the variable length of big int it is very easy to calculate the right `PUSH` function. I tried to change the type to `tosca.Word` but things got a little messy and unreadable.